explorer/interpreter/resolve_names.cpp

```diff
@@ -576,7 +576,9 @@ static auto ResolveNames(Declaration& declaration, StaticScope& enclosing_scope,
       auto& function = cast<CallableDeclaration>(declaration);
       StaticScope function_scope;
       function_scope.AddParent(&enclosing_scope);
-      enclosing_scope.MarkDeclared(function.name());
+      const auto name = GetName(function);
+      CARBON_CHECK(name) << "Unexpected missing name for `" << function << "`.";
+      enclosing_scope.MarkDeclared(std::string(*name));
```

Comment:
The double `std::string` constructor calls here are a bit of an awkward meeting between the API of `GetName` which returns a `std::string_view` and `MarkDeclared`/`MarkUsable` which takes a `std::string&`. Just noting it here.

Reply:
Agree, I don't see a simple way around that unless without making deep changes, I suppose that is acceptable.